packages/service-worker-mock/models/FetchEvent.js

```diff
@@ -2,19 +2,28 @@ const ExtendableEvent = require('./ExtendableEvent');
 const Request = require('./Request');
 
 class FetchEvent extends ExtendableEvent {
-  constructor(type, init) {
-    super();
-    this.type = type;
-    this.isReload = init.isReload || false;
-    if (init.request instanceof Request) {
-      this.request = init.request;
-    } else if (typeof init.request === 'string') {
-      this.request = new Request(init.request);
-    }
-  }
-  respondWith(response) {
-    this.promise = response;
-  }
+   constructor(type, init) {
+      super();
+      this.type = type;
+      this.isReload = init.isReload || false;
+      this.clientId = null;
+
+      let request = init.request;
+
+      if (typeof request === "object" && request.request) {
```

Comment:
According to the spec, init.request should only be a Request or a string. Can you move this logic into eventHandler.js instead?

Reply:
Ok I misunderstand what you wanted